SalesforceCommerceCloud/pwa-kit
# Allow support for multiple sites concurrently w/ `commerce-sdk-react`

![image](https://user-images.githubusercontent.com/8902581/213764277-0b4a056c-6bb4-4a7e-9e00-59ac1b74043a.png)

> Prefix your cookies with milk

# Description

To be in alignment with `plugin_slas` this PR ensures that the storage keys we use during the authentication process are prefixed with the id of the ecomm back-end we are connecting to. This means that you can jump from one site to another without loosing your guest or registered session, but more importantly the cookie keys will be the same format as what `plugin_slas` expects, so there will be no issues navigating from PWA to SFRA experiences.

# Changes

- Make BaseStorage an abstract class instead of an interface to add prefixing functionality.
- Remove clearing of storage as it's not in use anymore.
- Add new storage class for server side storage instead of using Maps directly (solves typing issue).

# How to Test-Drive This PR

- In the `test-commerce-sdk-react` project execute the command `npm start`.
- Goto the `slas login helpers` page. 
- Open your inspector and goto the application --> cookies tab and look at the cookies being created.
- You should see the `cc-nx-g` cookie prefixed with the currently selected site. 
- Try changing the site, you should see an additional cookie created with the newly selected site's id prefix to it.
- Now try logging in as a registered user, the guest cookie (`<prefix>_cc-nx-g`) will be replaces with the registered cooke (`<prefix>_cc-nx`)

# Checklists




## General
